federatedscope/core/monitors/metric_calculator.py

```diff
@@ -64,22 +64,27 @@ def _check_and_parse(self, ctx):
         if torch is not None and isinstance(y_prob, torch.Tensor):
             y_prob = y_prob.detach().cpu().numpy()
 
-        if y_true.ndim == 1:
-            y_true = np.expand_dims(y_true, axis=-1)
-        if y_prob.ndim == 2:
-            y_prob = np.expand_dims(y_prob, axis=-1)
-
-        y_pred = np.argmax(y_prob, axis=1)
-
-        # check shape and type
-        if not isinstance(y_true, np.ndarray):
-            raise RuntimeError('Type not support!')
-        if not y_true.shape == y_pred.shape:
-            raise RuntimeError('Shape not match!')
-        if not y_true.ndim == 2:
-            raise RuntimeError(
-                'y_true must be 2-dim arrray, {}-dim given'.format(
-                    y_true.ndim))
+        if 'regression' in ctx.cfg.model.task.lower():
+            y_pred = None
+        else:
+            # classification task
+            if y_true.ndim == 1:
+                y_true = np.expand_dims(y_true, axis=-1)
+            if y_prob.ndim == 2:
```

Comment:
I suggest to remove this augmentation and remove the traversal along with this expanded dimension in `eval_rmse()`

Reply:
We defer the (completed) support for multi-task setting to the next version. Currently, we'd better focus on single&multi-dimensional regression and single label binary/multi-class classification.

Reply:
For now, we add some comment to explain the usage of `y_true`, `y_prob`, and `y_pred`